cmd/zoekt-sourcegraph-indexserver/main.go

```diff
@@ -817,12 +800,19 @@ func newServer(conf rootConfig) (*Server, error) {
 	}
 
 	// Tune GOMAXPROCS to match Linux container CPU quota.
-	_, _ = maxprocs.Set()
+	_, err = maxprocs.Set()
+	if err != nil {
+		log.Printf("could not set GOMAXPROCS: %v\n", err)
+	}
 
 	// Set the sampling rate of Go's block profiler: https://github.com/DataDog/go-profiler-notes/blob/main/guide/README.md#block-profiler.
-	// The block profiler is disabled by default.
+	// The block profiler is disabled by default and should be enabled with care in production
 	runtime.SetBlockProfileRate(conf.blockProfileRate)
 
+	err = profiler.Init(ServiceName, zoekt.Version, conf.blockProfileRate)
```

Comment:
why did you move profiler.Init to here? Previously it happened in startServer. I believe that was intentional so we didn't capture short lived debug invocations of this binary.

Reply:
My thought here was to keep this close to the `runtime.SetBlockProfile`. If the Datadog tracer is enabled with the maximum number of profilers enabled it will redefine the above value. 

I suppose I don't fully understand _why_ we are setting the BlockProfileRate here. I'm lacking the context of why this would be set when not profiling or what affect this value would have before the change.

Reply:
zoekt-sourcegraph-indexserver can be run not as a daemon, but as a short lived command just for debug purposes. Those do not need datadog/cloud profiler and shouldn't have them.

I think you should move it to where we had it before. runtime.SetBlockProfile doesn't start the block profiler, that is why it was done where ti was before.